### PR TITLE
add support for jwt launches

### PR DESCRIPTION
**Pull request by rivernate**

### CODE REVIEW

File: spec/ims/lti/models/messages/message_spec.rb [hunk 1]
```diff
@@ -2,50 +2,53 @@
 
 module IMS::LTI::Models::Messages
   describe Message do
+
+    let(:message) {Message.new}
```
Comment (ebrohman) [inline spec/ims/lti/models/messages/message_spec.rb]: let( :message ) { described_class.new }

File: lib/ims/lti/models/messages/message.rb [hunk 2]
```diff
@@ -165,6 +165,26 @@ def method_missing(meth, *args, &block)
       end
     end
 
+    def to_jwt(private_key:, originating_domain:, algorithm: :HS256)
+      ims = post_params
```
Comment (abutterf) [inline lib/ims/lti/models/messages/message.rb]: Is this assignment really necessary? Also, where can I look at the spec to see how the payload should be formatted?

File: spec/ims/lti/models/messages/message_spec.rb [hunk 3]
```diff
@@ -152,33 +155,102 @@ class CustomMesage < Message
...
+
+      it "sets the iat to the current time" do
+        Timecop.freeze do
+          jwt = message.to_jwt(private_key: private_key, originating_domain: originating_domain )
+          json = JSON::JWT.decode(jwt, private_key)
+          expect(json[:iat]).to eq Time.now.to_i
+        end
+      end
+
+      it "sets the exp to the current time" do
```
Comment (abutterf) [inline spec/ims/lti/models/messages/message_spec.rb]: Shouldn't this be "to a future time"?

File: spec/ims/lti/models/messages/message_spec.rb (hunk 1 above)
Comment (rivernate) [inline spec/ims/lti/models/messages/message_spec.rb]: I updated let to a named subject, the problem I see with using "described_class" is that it creates a layer of abstraction  around what is going on. using `Message.new` is intention revealing, and less space

File: spec/ims/lti/models/messages/message_spec.rb (hunk 3 above)
Comment (rivernate) [inline spec/ims/lti/models/messages/message_spec.rb]: yup, thanks

File: lib/ims/lti/models/messages/message.rb (hunk 2 above)
Comment (rivernate) [inline lib/ims/lti/models/messages/message.rb]: I updated the last spec to be more explanatory

**Pull request merged**